Network based new account ban

S-S-X opened the pull request:
WIP, for visibility / comments.

Relies on geoip caching layer and prejoin hooks (to operate on incoming connections before creating new player accounts).
For simplicity uses pattern matching for IP checks.

S-S-X commented on create_account_ban.lua:
```diff
@@ -0,0 +1,185 @@
+	end
+end)
+
+local function kick_and_delete_player_account(name)
+	minetest.kick_player(name, MSG_BLOCKED)
+	local auth_handler = minetest.get_auth_handler()
+	if auth_handler.get_auth(name) then
+		minetest.log("action", "[beowulf] Removing player account: "..name)
+		auth_handler.delete_auth(name)
+		-- Temporary IP ban could be added here but probably not that useful, can be used to skip temp account creation
+	else
+		-- Should not happen, barrier is most probably leaking for some reason if this ever gets logged
+		minetest.log("error", "[beowulf] Player account should exist but not found: "..name)
+	end
+end
```
@BuckarooBanzay / @OgelGames Do you think this is safe enough or is there something that should be changed to make it better?

It is called when new accounts are blocked based on IP/ASN, this is workaround to allow enforcing new account ban with async geoip lookup.

Automated tests would be good (and I'll include some) to prevent account deletion because of some stupid mistake but besides that for actual implementation?

BuckarooBanzay replied:
> @BuckarooBanzay / @OgelGames (or anyone else) Do you think this is safe enough or is there something that should be changed to make it better?

LGTM, the kick message should be clear enough i think :+1:

S-S-X commented:
Cleaned up a bit and rebased to master, still no tests however.
Probably some day in future I'll either do some manual testing or write some automated tests for this... I think most important thing is to make sure that `kick_and_delete_player_account` wont be accidentally triggered and that it wont remove any accounts that were created before ongoing authentication sequence.

S-S-X commented:
Readme changes: 
* removed "_State: **WIP**_" (it is fully functional and tested mod).
* added new configuration key (one introduced in this PR).
* converted configuration section to table.

S-S-X reviewed (commented):
Pattern type selection and pattern matching are both broken, actually just reading code should easily make that clear as there's few very simple mistakes and also no pattern escaping making patterns way too complicated.

- [ ] IP match patterns should allow only subset of full Lua patterns.
- [ ] ASN should allow more than just single digit...
- [ ] ASN should probably check validity against root ASN chunk allocations (and either warning or error when private/test numbers given).
- [ ] `-logging` does not do anything, I don't even remember anymore what I was supposed to do with it...
- [ ] TBD: chat command name, currently `/block <param>`. Should that be changed to something else?

BuckarooBanzay commented:
>  TBD: chat command name, currently /block <param>. Should that be changed to something else?

maybe prefix it like the other mods do with `/beowulf_block <param>` ?
